agent: don't wait for autoconfig to complete in DEV_SET_CONFIG

arnout opened the pull request:
For wireless backhaul, this is incorrect because there is no backhaul yet on which to perform autoconfig.
For wired backhaul, it is not needed nor expected by UCC.

arnout committed:
agent: backhaul_manager: remove EOL whitespace in comments

clang-format doesn't clean whitespace in comments. However, some editors
do clean whitespace when saving. Clean whitespace now.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
agent: ucc listener: remove EOL whitespace in comments

clang-format doesn't clean whitespace in comments. However, some editors
do clean whitespace when saving. Clean whitespace now.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
agent: UCC listener: dev_set_config: don't wait for autoconfig

When the dev_set_config command was implemented for the agent UCC
listener, we thought that it should wait for AP autoconfig to complete
before returning. However:

- for wired backhaul, it's not really needed because it is always
  followed by a 5 second sleep, plenty of time for onboarding to finish;
- for wireless backhaul, it's not possible because there is no backhaul
  connection until WPS is done, so it never completes.

Therefore, we can simply return immediately from handle_dev_set_config
rather than waiting for onboarding success.

This also allows us to vastly simplify the onboarding state handling.
Rather than exposing the onboarding state to the backhaul manager, we
can simply replace it by the is_in_reset() check. We still do need the
WAIT_FOR_RESET state to make sure DEV_RESET_DEFAULT doesn't return until
the reset is complete, but all other states can be replaced by a single
state CONFIGURED.

This in turns allows us to simplify the onboarding state handling in the
backhaul manager thread. Instead of checking and "advancing" the state
in different places, we can simply check at the beginning of
backhaul_fsm_main if we should stay in reset. Handling of
ACTION_BACKHAUL_ONBOARDING_FINISHED_NOTIFICATION is no longer needed.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
agent: remove ACTION_BACKHAUL_ONBOARDING_FINISHED_NOTIFICATION

Since it is no longer needed for UCC onboarding state, we can remove
this message from the flow.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout committed:
btlvf: remove ACTION_BACKHAUL_ONBOARDING_FINISHED_NOTIFICATION

It is no longer used so the message can be removed.

Update AutoGenerated as well.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

tomereli reviewed (approved):
Thank you for simplifying this horrible onboarding states👌

arnout committed:
agent: UCC listener: add explicit reset_completed() function

When DEV_RESET_DEFAULT is done, we currently return immediately, before
the reset has actually completed (i.e. before the slaves are ready). Up
to now, this wasn't a problem because DEV_SET_CONFIG will wait until
onboarding is done. However, we changed DEV_SET_CONFIG to return
immediately. Tests will wait for a few seconds after DEV_SET_CONFIG for
onboarding to complete, but the slaves may take a lot longer to become
ready.

In addition. when DEV_RESET_DEFAULT finished, we should be sure that the
agent really is reset. Currently we return before the RESTART state is
reached.

Therefore, track the reset state explicitly in the UCC listener. The
backhaul will signal reset_completed() as soon as it reaches ENABLED
state.

To simplify things, keep track of the state as two separate booleans:
m_in_reset and m_reset_completed.

Certification tests should still pass: MAP-4.3.2_ETH_FH24G:netgear-rax40

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

rmelotte commented on agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:
```diff
@@ -763,6 +753,23 @@ bool backhaul_manager::backhaul_fsm_main(bool &skip_select)
 
+    // UCC FSM. If UCC is in RESET, we have to stay in (or move to) ENABLED state.
+    if (m_agent_ucc_listener) {
+        if (m_agent_ucc_listener->is_in_reset()) {
```
```suggestion
    if (m_agent_ucc_listener && m_agent_ucc_listener->is_in_reset()) {
```
